libp2p/switch.nim

```diff
@@ -364,12 +377,11 @@ proc start*(s: Switch) {.public, async: (raises: [CancelledError, LPError]).} =
 
   for t in s.transports: # for each transport
     if t.addrs.len > 0 or t.running:
+      if t of TcpTransport:
+        continue # already added previously
```

Comment:
please explain

Reply:
hmm.. could this code be changed:

```
      s.acceptFuts.add(s.accept(t))
      s.peerInfo.listenAddrs &= t.addrs
```

so that it only happens once? 
if this means we need to sort items in specific order, lets sort items then process them in that order in one place rather then two places.

Reply:
aha i get it now....